### DISCUSSION TITLE
The resource could not be resolved

### DISCUSSION THREAD

**Discussion by erotavlas** [MahApps/MahApps.Metro#4084] [Q&A] [answered]
I just upgraded my app from MahApps 1.6.0 to the latest version 2.4.4 and I'm trying to resolve the following issue.  

I have multiple location in my app where I create Styles basedon some Metro styles For example

```
    xmlns:Controls="http://metro.mahapps.com/winfx/xaml/controls"

    <Controls:MetroWindow.Resources>
        <local:BindingProxy x:Key="Proxy" Data="{Binding}"></local:BindingProxy>
        <Style x:Key="TabItemStyle" TargetType="TabItem" BasedOn="{StaticResource MetroTabItem}">
            <Setter Property="IsEnabled" Value="{Binding isTabItemEnabled}" />
        </Style>
    </Controls:MetroWindow.Resources>
```

However I can no longer build the project and I tried to replace the based on with the following

`<Style x:Key="TabItemStyle" TargetType="TabItem" BasedOn="{StaticResource {x:Type Controls:MetroTabItem}}">
`

But no matter what I try it cannot resolve these items anymore

EDIT: I forgot to mention the problem I'm also finding is runtime, I'm getting errors like this

`Exception: Cannot find resource named 'MetroMenuItem'. Resource names are case sensitive.`

Any ideas?

**Comment by timunie** [comment, marked as answer]
Hi

Maybe this link helps to find the new Keys. https://mahapps.com/docs/guides/migration-to-v2.0 

You can also use `BasedOn {StaticResource {x:Type WhatEverControl}}`

Happy coding
Tim

**Reply by erotavlas** [reply]
Ok thank you! The link you provided helped and I was able to get my app running again :)
